darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1001,9 +1001,12 @@ def _setup_for_train(
                 collate_fn=self._batch_collate_fn,
             )
         )
-
-        # if user wants to train the model for more epochs, ignore the n_epochs parameter
-        train_num_epochs = epochs if epochs > 0 else self.n_epochs
+        # Fetch max_epochs to pass to the trainer
+        train_num_epochs = (
+            max(self.epochs_trained, epochs)
+            if epochs
```

Comment:
if epochs is smaller than self.epochs_trained, it means that the model won't be further trained, right? From my understanding, this PR was supposed to make so that calling `fit()` on a already trained model would "incrementally retrain" it for the number of epochs indicated.

Reply:
> I have the impression that the PR is not solving the problem described in the associated issue(s). Correct me if I am wrong but I think that the intended behavior is that regardless of the number of `epochs_trained`, when `fit()` is called on a model, it will be trained for this number of additional epochs (and not "until this number of epochs is reached", so that the user doesn't have to remember for how many epochs the model was trained.
> 
> On a side note, I think that this PR also fixes #1090 (which describe the exact same issue as #1495).

Well, this PR:
- fixes the exsting issue with resuming training from checkpoints;
- but leaves the rest of the logic (regarding how we handle the 'additional' number of epochs passed in the fit call) unchanged

I'm all in favour of making this change though. It seems to me this is indeed more intuitive. I'll incorporate it in this PR

Reply:
@madtoinou updated the PR